tlsn/benches/src/lib.rs

```diff
@@ -1,6 +1,259 @@
+pub mod config;
+pub mod metrics;
+
+use std::{io, process::Command};
+
 pub const PROVER_NAMESPACE: &str = "prover-ns";
 pub const PROVER_INTERFACE: &str = "prover-veth";
 pub const PROVER_SUBNET: &str = "10.10.1.0/24";
 pub const VERIFIER_NAMESPACE: &str = "verifier-ns";
 pub const VERIFIER_INTERFACE: &str = "verifier-veth";
 pub const VERIFIER_SUBNET: &str = "10.10.1.1/24";
+
+pub fn set_up() -> io::Result<()> {
+    // Create network namespaces
+    create_network_namespace(PROVER_NAMESPACE)?;
+    create_network_namespace(VERIFIER_NAMESPACE)?;
+
+    // Create veth pair and attach to namespaces
+    create_veth_pair(
+        PROVER_NAMESPACE,
+        PROVER_INTERFACE,
+        VERIFIER_NAMESPACE,
+        VERIFIER_INTERFACE,
+    )?;
+
+    // Set devices up
+    set_device_up(PROVER_NAMESPACE, PROVER_INTERFACE)?;
+    set_device_up(VERIFIER_NAMESPACE, VERIFIER_INTERFACE)?;
+
+    // Assign IPs
+    assign_ip_to_interface(PROVER_NAMESPACE, PROVER_INTERFACE, PROVER_SUBNET)?;
+    assign_ip_to_interface(VERIFIER_NAMESPACE, VERIFIER_INTERFACE, VERIFIER_SUBNET)?;
+
+    // Set default routes
+    set_default_route(
+        PROVER_NAMESPACE,
+        PROVER_INTERFACE,
+        PROVER_SUBNET.split('/').nth(0).unwrap(),
+    )?;
+    set_default_route(
+        VERIFIER_NAMESPACE,
+        VERIFIER_INTERFACE,
+        VERIFIER_SUBNET.split('/').nth(0).unwrap(),
+    )?;
+
+    Ok(())
+}
+
+pub fn clean_up() {
+    // Delete interface pair
+    if let Err(e) = Command::new("sudo")
```

Comment:
The bench is already running as root, so this (and the next) `sudo` are not necessary.